docs/deep-linking-android.md

```diff
@@ -0,0 +1,91 @@
+Deep-linking using Circuit
+==========================
+
+Deep linking to application is a vast topic and the implementation detail can vary based on your application needs.
+Deep linking strategy can be highly sophisticated or simple based on the use-case.
+
+To keep things very simple and easy to understand, we will be focusing on deep linking to **Android** platform only. 
+You can then extend the idea to other platforms as needed.
+
+!!! info
+    Before you begin, we recommend having a basic understanding of deep linking. To get the most out of this guide, please review Android's official [training guide](https://developer.android.com/training/app-links/deep-linking), which provides valuable insights into deep linking for Android apps.
+
+### Deep linking strategy
+Essentially, you need to define a strategy for you app to handle the incoming deep link. We will take a look at Circuit's [email app](https://slackhq.github.io/circuit/tutorial/) from the tutorial with following screens:
+
+* **Inbox Screen** - List all incoming emails
+* **Details Screen** - Show details of the email
+* **Draft Screen** - Compose a new email _(additional screen added for this demo)_
+
+By default, the app launches with **Inbox Screen** as the root screen. To deep link into other two screens, we will define URI path segment as:
+
+* **`inbox`** - Inbox Screen
+* **`view_email`** - Details Screen with `emailId` as query parameter
+* **`new_email`** - Draft Screen
+
+Assuming, our app's URI scheme is `circuitapp://` with host `emailonthego`, the deep links will look like:
+
+* `circuitapp://emailonthego/inbox` - Deep link to Inbox Screen
+* `circuitapp://emailonthego/inbox/view_email?emailId=2` - Deep link to Details Screen with email id 2 with Inbox Screen as parent
+* `circuitapp://emailonthego/inbox/new_email` - Deep link to Draft Screen with Inbox Screen as parent
+* `circuitapp://emailonthego/inbox/view_email/new_email?emailId=3` - Deep link to Draft Screen and in backstack there is Details Screen and Inbox Screen.
+
+Once you have added [intent-filter](https://developer.android.com/training/app-links/deep-linking#adding-filters) in your `AndroidManifest.xml` file, you can handle the incoming deep link in your `Activity`.
+
+### Handling deep link in Circuit
+In our case, we will create a `parseDeepLink` function to parse the incoming deep link and return the list of screens to be displayed in the app.
+```kotlin
+// In your `MainActivity.kt` or activity that has the `intent-filter` for custom URI scheme
+override fun onCreate(savedInstanceState: Bundle?) {
+    // ...
+
+    setContent {
+        ComposeAppTheme {
```

Comment:
shouldn't this be `CircuitAppTheme`?

Reply:
Updated to use the `MaterialTheme` to match the tutorial [sample](https://github.com/slackhq/circuit/blob/main/samples/tutorial/src/androidMain/kotlin/com/slack/circuit/tutorial/impl/MainActivityImpl.kt#L25).